u-root.go

```diff
@@ -335,21 +108,72 @@ func Main() error {
 		baseFile = uroot.DefaultRamfs.Reader()
 	}
 
-	initCommand := *initCmd
-	if *fourbins && *build == "source" {
-		initCommand = "/go/bin/go"
+	tempDir := *tmpDir
+	if tempDir == "" {
+		var err error
+		tempDir, err = ioutil.TempDir("", "u-root")
+		if err != nil {
+			return err
+		}
+		defer os.RemoveAll(tempDir)
+	} else if _, err := os.Stat(tempDir); os.IsNotExist(err) {
+		if err := os.MkdirAll(tempDir, 0755); err != nil {
+			return fmt.Errorf("temporary directory %q did not exist; tried to mkdir but failed: %v", tempDir, err)
+		}
 	}
 
-	opts := uroot.Opts{
-		Env: env,
+	var (
+		c           []uroot.Commands
+		initCommand = *initCmd
+	)
+	if !*noCommands {
```

Comment:
What was the reason behind adding this flag? I doesn't seem needed by the rest of the PR.

Reply:
to be able to test the -extra-files commands without having to compile anything. speeds up the tests.